cmake/FetchExternalDeps.cmake

```diff
@@ -1,5 +1,9 @@
 Include(FetchContent)
 
```

Comment:
Maybe only fetch Catch2 if testing is enabled?
`if(BUILD_TESTING)`

Reply:
oh yeah.. sure.. 👍